lib/chef/provider/apt_repository.rb

```diff
@@ -51,6 +51,19 @@ def load_current_resource
           end
         end
 
+        # make sure we have apt-transport-https installed before we try to
+        # apt-get update with a https repo defined
+        if new_resource.uri =~ /^https/
+          declare_resource(:apt_update, new_resource.name) do
+            ignore_failure true
+            action :periodic
+          end
```

Comment:
still think we should convert this over into a PR to talk about adding that as a feature to apt_package in general tho....